src/Microsoft.VisualStudio.ProjectSystem.Managed.TestServices/ProjectSystem/ProjectTreePropertiesProviderExtensions.cs

```diff
@@ -15,13 +16,23 @@ internal static class ProjectTreePropertiesProviderExtensions
         /// </summary>
         public static IProjectTree ChangePropertyValuesForEntireTree(this IProjectTreePropertiesProvider propertiesProvider, IProjectTree tree)
```

Comment:
Similar to Roslyn, manipulations of IProjectTree are always done in an immutable manner - ie when our product code manipulates a tree, it's always returning a new value. Any method that takes a IProjectTree and returns one, is always creating a "new" tree if it changed any value, trees are supposed to be immutable. We're basically following the spirit of the rest of CPS - by returning a new value. We're doing this "hack" because I've implemented a test IProjectTree that's breaking immutability rules by modifying itself on any manipulation methods (such as `IProjectTree SetIcon(...)`) - it would be great to fix that, but its probably a least a couple of days of work - and probably not worth it for test code.

This method name is named after the actual method that it's calling on IProjectTreePropertiesProviders.ChangePropertyValues, to better indicate what's it's doing, to change it would hide that.